Piwik.Tracker/PiwikTracker.cs

```diff
@@ -1160,14 +1159,15 @@ public void SetIp(string ip)
         /// <exception cref="Exception"/>
         public void SetUserId(string userId)
         {
+            // todo this is contradicion: (userId == null) / string.IsNullOrEmpty(userId)

```

Comment:
Could you elaborate on this comment?

This is what I understand : 

 - when calling `SetUserId(null)`, a new visitor id will be generated
 - when calling `SetUserId("")`, an exception is raised

It seems to be consistent with the behavior in the PHP lib : https://github.com/piwik/piwik-php-tracker/blob/master/PiwikTracker.php#L1167